### PR TITLE
MOTECH-2356: Added getRelationship by person UUID to the OpenMRS data…

### PR DESCRIPTION
**Pull request by ngraczewski**
… provider

### CODE REVIEW

File: openmrs-19/src/main/java/org/motechproject/openmrs19/service/OpenMRSRelationshipService.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    /**
+     * Returns relationship of the type with the given {@code typeUuid} for the person with the given
+     * {@code personUuid}.
+     *
+     * @param configName  the name of the configuration to be used while performing this action
+     * @param typeUuid  the UUID of the type
+     * @param personUuid  the UUID of the person
+     * @return the matching relationship
+     */
+    Relationship getByTypeUuidAndPersonUuid(String configName, String typeUuid, String personUuid);
```
Comment (pgesek) [inline openmrs-19/src/main/java/org/motechproject/openmrs19/service/OpenMRSRelationshipService.java]: Shouldn't we also have a version with no config name, that uses the default config, just like in Commcare?

File: openmrs-19/src/main/java/org/motechproject/openmrs19/service/impl/OpenMRSRelationshipServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+                if (relationship.getPersonB() != null) {
+                    personBUuid = relationship.getPersonB().getUuid();
+                }
+
+                if (relationship.getRelationshipType() != null) {
+                    relationshipTypeUuid = relationship.getRelationshipType().getUuid();
+                }
+
+                return StringUtils.equals(personBUuid, personUuid)
```
Comment (pgesek) [inline openmrs-19/src/main/java/org/motechproject/openmrs19/service/impl/OpenMRSRelationshipServiceImpl.java:48]: why are we checking the person uuid here? won't the fetch handle that, given that we already pass the uuid as aparameter?

File: openmrs-19/src/main/java/org/motechproject/openmrs19/service/impl/OpenMRSRelationshipServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+                        && StringUtils.equals(relationshipTypeUuid, typeUuid);
+            }
+        }));
+
+        if (relationships.size() > 1) {
```
Comment (pgesek) [inline openmrs-19/src/main/java/org/motechproject/openmrs19/service/impl/OpenMRSRelationshipServiceImpl.java]: is it out of place to have two sons for example? Or am I getting this wrong? If this is done stricly for tasks reasons, then this check belongs in the data provider class

File: openmrs-19/src/test/java/org/motechproject/openmrs19/tasks/OpenMRSTaskDataProviderTest.java [hunk 4]
```diff
@@ -264,4 +276,48 @@ public void shouldReturnProviderForLookupGetProviderByUuid() {
         assertEquals(provider, object);
         verify(providerService).getProviderByUuid(null, "5");
     }
+
+    @Test
+    public void shouldReturnRelationshipForLookupGetRelationshipByTypeUuidAndPersonBUuid() {
```
Comment (pgesek) [inline openmrs-19/src/test/java/org/motechproject/openmrs19/tasks/OpenMRSTaskDataProviderTest.java:283]: No IT test? We can create a separate ticket for it at the very least

File: openmrs-19/src/main/java/org/motechproject/openmrs19/service/OpenMRSRelationshipService.java (hunk 1 above)
Comment (ngraczewski) [inline openmrs-19/src/main/java/org/motechproject/openmrs19/service/OpenMRSRelationshipService.java]: Methods without config name as parameters have been removed. With @kmadej changes we're losing backward compatibility, so those methods became redundant. I couldn't think of any other use case for them.

File: openmrs-19/src/main/java/org/motechproject/openmrs19/service/impl/OpenMRSRelationshipServiceImpl.java (hunk 2 above)
Comment (ngraczewski) [inline openmrs-19/src/main/java/org/motechproject/openmrs19/service/impl/OpenMRSRelationshipServiceImpl.java:48]: As mentioned in the [MOTECH-2356](https://applab.atlassian.net/browse/MOTECH-2356), at this moment some of the relationship filters doesn't work on the OpenMRS server. As a temporary workaround, we're filtering relationships by its "person", which includes both personA and personB fields. We're interested only in the personB, hence we are checking that person UUID.

File: openmrs-19/src/main/java/org/motechproject/openmrs19/service/impl/OpenMRSRelationshipServiceImpl.java (hunk 3 above)
Comment (ngraczewski) [inline openmrs-19/src/main/java/org/motechproject/openmrs19/service/impl/OpenMRSRelationshipServiceImpl.java]: It was done strictly for task reasons. The check has been moved to the data provider class.

File: openmrs-19/src/test/java/org/motechproject/openmrs19/tasks/OpenMRSTaskDataProviderTest.java (hunk 4 above)
Comment (ngraczewski) [inline openmrs-19/src/test/java/org/motechproject/openmrs19/tasks/OpenMRSTaskDataProviderTest.java:283]: Whole OpenMRSTaskDataProvider could use some integration tests. I've created ticket for it [MOTECH-2431](https://applab.atlassian.net/browse/MOTECH-2431).

**Pull request merged**